tsdb/meta.go

```diff
@@ -1772,9 +1772,7 @@ func (s stringSet) intersect(o stringSet) stringSet {
 // MeasurementFromSeriesKey returns the name of the measurement from a key that
 // contains a measurement name.
 func MeasurementFromSeriesKey(key string) string {
-	idx := strings.Index(key, ",")
-	if idx == -1 {
-		return key
-	}
-	return key[:idx]
+	// Ignoring the error because the func returns "missing fields"
+	k, _, _ := models.ParseKey(key)
+	return escape.UnescapeString(k)
```

Comment:
No, it has always been broken, but never discovered.  This is called to parse a key already on disk.